assets/shaders/gpu_readback.wgsl

```diff
@@ -0,0 +1,6 @@
+@group(0) @binding(0) var<storage, read_write> data: u32;
+
+@compute @workgroup_size(1, 1, 1)
+fn main() {
+    data += 1u;
```

Comment:
I might like to see a more interesting example than this but I guess the interesting part of this isn't the shader itself. It should probably at least be a vec though since that's the normal use case.

Reply:
a vec example would be nice. following this example, i'm not sure how to handle the `BUFFER_SIZE` with a vec.

Reply:
I didn't really want this example to be a way to teach how to use compute shaders, but I guess having an example that's slightly more realistic makes more sense. 

@stowmyy using BUFFER_SIZE with a vec you'd just need to get the `len * size in bytes of each element`. A vec with a single item would have the same size here for example.

Reply:
yes! i assume the vec should always have a fixed size though, resizing the buffer doesn't sound like a good idea? that's more what i meant about not being sure how to handle it.

otherwise i think this example is great already.

Reply:
I believe arrays in storage buffers can have arbitrary length, but not all devices support that. You do need to know the len at the time of reading or writing it though. I used a const here for simplicity but it's not mandatory. Having a varying length would make it harder to figure out how you want to split your compute shader work groups though.